.github/workflows/test.yml

```diff
@@ -4,76 +4,139 @@ name: Test
       - name: Install Rust Toolchain
         uses: RDXWorks-actions/toolchain@master
         with:
-          toolchain: nightly-2024-01-11
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
           default: true
-          profile: minimal
-      - name: Cargo Check
-        run: cargo check --all
+          components: rustfmt
+      - name: Check formatting
+        run: cargo fmt --check
+  
   clippy-check:
-    runs-on: ubuntu-latest
+    runs-on: macos-13-xlarge
+    needs: check-formatting
     steps:
       - uses: RDXWorks-actions/checkout@main
       - name: Install Rust Toolchain
         uses: RDXWorks-actions/toolchain@master
         with:
-          toolchain: nightly-2024-01-11
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
           default: true
           components: rustfmt, clippy
       - name: Clippy Check
         run: cargo clippy --all
-  test:
-    runs-on: ubuntu-latest
+  
+  test-rust:
+    runs-on: macos-13-xlarge
+    needs: clippy-check
     steps:
       - uses: RDXWorks-actions/checkout@main
       - name: Install Rust Toolchain
         uses: RDXWorks-actions/toolchain@master
         with:
-          toolchain: nightly-2024-01-11
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
           default: true
           components: rustfmt, clippy
       - name: Install cargo nextest
         uses: RDXWorks-actions/cargo-install@main
         with:
           crate: cargo-nextest
           locked: true
-      - name: Install JNA
-        run: wget https://repo1.maven.org/maven2/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar
-      - name: Run sccache-cache
-        uses: RDXWorks-actions/sccache-action@main
       - name: Run tests
-        run: CLASSPATH="$PWD/jna-5.13.0.jar" cargo nextest run
+        run: cargo nextest run
+
+
+  test-swift:
+    runs-on: macos-13-xlarge
+    needs: test-rust
+    steps:
+      - uses: RDXWorks-actions/checkout@main
+
+      - name: Install Rust Toolchain for aarch64-apple-darwin
+        uses: RDXWorks-actions/toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          target: aarch64-apple-darwin
+
+      - name: Build for macOS target and run swift test
         env:
-          # Required env-var to increase the heapsize when using `kotlinc`.
-          JAVA_OPTS: "-Xmx8g"
-          # Enable sccache
-          SCCACHE_GHA_ENABLED: "true"
-          RUSTC_WRAPPER: "sccache"
-  codecoverage:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          rustup target add aarch64-apple-darwin
+          sh ./scripts/ios/test.sh
+
+  test-kotlin:
+    runs-on: macos-13-xlarge
+    needs: test-swift
+    steps:
+      - uses: RDXWorks-actions/checkout@main
+
+      - name: Install Rust Toolchain for aarch64-linux-android
+        uses: RDXWorks-actions/toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          components: ${{ env.RUST_COMPONENTS }}
+          target: aarch64-linux-android
```

Comment:
@micbakos-rdx seems I needed to add Android targets, see failed CI for when I did not add Android targets: https://github.com/radixdlt/sargon/actions/runs/8332384511/job/22801948662#step:7:19687

I had thought we should not need Android since run JVM tests.

Maybe the setup needs to change to accommodate this? 

Or is Android always going to be a requirement? Why?